examples/all-clusters-app/nxp/mw320/binding-handler.cpp

```diff
@@ -0,0 +1,141 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "binding-handler.h"
+
+#include "app-common/zap-generated/ids/Clusters.h"
+#include "app-common/zap-generated/ids/Commands.h"
+#include "app/CommandSender.h"
+#include "app/clusters/bindings/BindingManager.h"
+#include "app/server/Server.h"
+#include "controller/InvokeInteraction.h"
+#include "lib/core/CHIPError.h"
+#include "platform/CHIPDeviceLayer.h"
+
+#if defined(ENABLE_CHIP_SHELL)
+#include "lib/shell/Engine.h"
+
+using chip::Shell::Engine;
+using chip::Shell::shell_command_t;
+using chip::Shell::streamer_get;
+using chip::Shell::streamer_printf;
+#endif // defined(ENABLE_CHIP_SHELL)
+
+static bool sSwitchOnOffState = false;
+#if defined(ENABLE_CHIP_SHELL)
+static void ToggleSwitchOnOff(bool newState)
+{
+    sSwitchOnOffState = newState;
+    chip::BindingManager::GetInstance().NotifyBoundClusterChanged(1, chip::app::Clusters::OnOff::Id, nullptr);
+}
+
+static CHIP_ERROR SwitchCommandHandler(int argc, char ** argv)
+{
+    if (argc == 1 && strcmp(argv[0], "on") == 0)
+    {
+        ToggleSwitchOnOff(true);
+        return CHIP_NO_ERROR;
+    }
+    if (argc == 1 && strcmp(argv[0], "off") == 0)
+    {
+        ToggleSwitchOnOff(false);
+        return CHIP_NO_ERROR;
+    }
+    streamer_printf(streamer_get(), "Usage: switch [on|off]");
+    return CHIP_NO_ERROR;
+}
+
+static void RegisterSwitchCommands()
+{
+    static const shell_command_t sSwitchCommand = { SwitchCommandHandler, "switch", "Switch commands. Usage: switch [on|off]" };
+
+    Engine::Root().RegisterCommands(&sSwitchCommand, 1);
+    return;
+}
+#endif // defined(ENABLE_CHIP_SHELL)
+
+static void BoundDeviceChangedHandler(const EmberBindingTableEntry & binding, chip::DeviceProxy * peer_device, void * context)
+{
+    using namespace chip;
+    using namespace chip::app;
+
+    if (binding.type == EMBER_MULTICAST_BINDING)
+    {
+        ChipLogError(NotSpecified, "Group binding is not supported now");
+        return;
+    }
+
+    if (binding.type == EMBER_UNICAST_BINDING && binding.local == 1 &&
+        (!binding.clusterId.HasValue() || binding.clusterId.Value() == Clusters::OnOff::Id))
+    {
+        auto onSuccess = [](const ConcreteCommandPath & commandPath, const StatusIB & status, const auto & dataResponse) {
+            ChipLogProgress(NotSpecified, "OnOff command succeeds");
+        };
+        auto onFailure = [](CHIP_ERROR error) {
+            ChipLogError(NotSpecified, "OnOff command failed: %" CHIP_ERROR_FORMAT, error.Format());
+        };
+        // Note: Need to change using toggle command since sSwitchOnOffState won't be changed if not triggerred from the switch
+        // command (SwitchCommandHandler)
+#if 0
```

Comment:
Delete commented out code

Reply:
Or use some global define with a description of what gets enabled/disabled. `#if 0` is too vague and is a commenting out of code. In this case it looks like you want a compile time define choice.

Reply:
Done with commit https://github.com/project-chip/connectedhomeip/commit/5b1f44cfece69292b7397fdcf6951f31de44e1c9.

Reply:
Please let me know if I missed anything. Thanks.

Reply:
Thanks for your help.